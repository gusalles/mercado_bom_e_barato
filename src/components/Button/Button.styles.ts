import { Button, styled } from '@mui/material';

export const StyledButton = styled(Button)(() => ({
  marginTop: '16px',
  borderRadius: '12px',
  color: '#000',
  textTransform: 'initial',
  fontWeight: 600,
  fontSize: '1rem',
  height: '56px',
}));
