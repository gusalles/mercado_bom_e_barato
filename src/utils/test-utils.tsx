import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
