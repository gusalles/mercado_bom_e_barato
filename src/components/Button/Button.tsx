import { ButtonProps } from '@mui/material';
import { StyledButton } from './Button.styles';

export function Button(props: ButtonProps) {
  return <StyledButton variant="outlined" {...props} />;
}
