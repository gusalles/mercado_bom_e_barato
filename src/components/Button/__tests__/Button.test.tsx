import { render, screen } from '@/utils/test-utils';
import { Button } from '../Button';

describe('Button', () => {
  it('should render', () => {
    render(<Button />);

    const button = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });
});
