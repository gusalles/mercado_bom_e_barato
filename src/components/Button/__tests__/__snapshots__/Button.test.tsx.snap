// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-gtbsfe-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
/>
`;
